src/lib/datepicker/datepicker.spec.ts

```diff
@@ -433,6 +433,58 @@ describe('MdDatepicker', () => {
         expect(testComponent.datepicker._minDate).toEqual(new Date(2010, JAN, 1));
         expect(testComponent.datepicker._maxDate).toEqual(new Date(2020, JAN, 1));
       });
+
+      it('should mark input invalid', async(() => {
```

Comment:
can you split this into separate tests?
```
'should mark invalid when value is before min'
'should mark invalid when value is after max'
'should not mark invalid when value equals min'
'should not mark invalid when value equals max'
'should not mark invalid when value is between min and max'
```
this way we avoid the deep nesting and if we get a failure its easier to see why